api.go

```diff
@@ -875,10 +875,14 @@ func (c *Client) makeTargetURL(bucketName, objectName, bucketLocation string, is
 	if h, p, err := net.SplitHostPort(host); err == nil {
 		if scheme == "http" && p == "80" || scheme == "https" && p == "443" {
 			host = h
+			if ip := net.ParseIP(h); ip != nil {
```

Comment:
I think what is missing here is to check if this is an ipv6 address, so we need just another if

Reply:
that is already added ParseIP parses IPv4 and IPv6 and then ip.To16() says that its IPv6 @vadmeste

Reply:
> that is already added ParseIP parses IPv4 and IPv6 and then ip.To16() says that its IPv6 @vadmeste

what I am saying is that there is no ip.To16() in your code right now

Reply:
done fixed.